lib/hal/video.c

```diff
@@ -310,67 +335,28 @@ void XVideoInit(DWORD dwMode, int width, int height, int bpp)
 
 BOOL XVideoSetMode(int width, int height, int bpp, int refresh)
 {
-	VIDEO_MODE_SETTING *pVidMode = NULL;
-	int vidRefresh = 0;
-	int i = 0;
-
-	DWORD dwEnc = XVideoGetEncoderSettings();
-	
-	DWORD dwAdapter		= dwEnc & 0x000000FF;
-	DWORD dwStandard	= dwEnc & 0x0000FF00;
-
-	if(bpp != 15 && bpp != 16 && bpp != 32)
-		bpp = 32;
-
-	if(refresh > 0)
-		vidRefresh = refresh;
-	else
-	{
-		if(dwEnc & 0x00400000) //60Hz refresh rate
-			vidRefresh = 60;
-		else
-			vidRefresh = 50;
-	}
+	VIDEO_MODE vm;
+	void *p = NULL;
 
-	for(i=0; i<iVidModes; i++)
+	while(XVideoListModes(&vm, bpp, refresh, &p))
 	{
-		pVidMode = &vidModes[i];
-
-		if((pVidMode->dwFlags & 0x000000FF) != dwAdapter)
-			continue;
-
-		if(pVidMode->dwStandard != dwStandard)
-			continue;
-
-		if(pVidMode->width != width || pVidMode->height != height)
+		if(vm.width != width || vm.height != height)
+		{
 			continue;
+		}
```

Comment:
nit: existing code (old code which this replaces + code in `XVideoListModes`) do not have `{}` for these early-rejections

Reply:
Fixed - curly braces removed.